cloud/cluster/fetch_cron.go

```diff
@@ -0,0 +1,38 @@
+package cluster
+
+import (
+	"fmt"
+	"time"
+)
+
+type FetchCron struct {
+	Ticker  *time.Ticker
+	f 	    Fetcher
+	Cl 		*Cluster
+}
+
+// Returns a full list of visible nodes.
+type Fetcher interface {
+	Fetch() ([]Node, error)
+}
+
+func NewFetchCron(f Fetcher, t time.Duration, cl *Cluster) *FetchCron {
```

Comment:
Maybe just pass the chan instead of the whole cluster.

(FWIW, this is an application of the https://en.wikipedia.org/wiki/Law_of_Demeter )